elastic/kibana
# [Console] Add a setting to disable the a11y overlay

## Summary
Fixes https://github.com/elastic/kibana/issues/16139

This PR adds a setting in Console that allows to disable the a11y overlay. The default will be set to `enabled` so that the default behaviour should not change. A user can disable the overlay in their browser and the setting is saved in local storage. So that other users are not affected by that change. 

The reason to allow disabling the overlay is that it can be flaky and sometimes it's displayed when not intended. The code relies on `querySelector` (see this [file](https://github.com/elastic/kibana/blob/1b3f23829c9a41d2995e279b37b04fc457befa6e/src/plugins/es_ui_shared/__packages_do_not_import__/ace/use_ui_ace_keyboard_mode.tsx#L25)) so I think that causes the flakiness and that is very difficult to test reliably. 

### Screenshot 
#### A11y overlay (no changes)
<img width="1483" alt="Screenshot 2023-06-01 at 16 34 23" src="https://github.com/elastic/kibana/assets/6585477/d776625c-92cd-4bd9-8e5e-2f672df351a4">

#### Settings modal with the new option to disable the a11y overlay
<img width="474" alt="Screenshot 2023-06-01 at 16 29 02" src="https://github.com/elastic/kibana/assets/6585477/8745c7a0-62f4-41a9-9eff-ff8bebd4f767">


#### How to test 
1. Start Kibana and navigate to the Console
2. Press ESC when textarea is focused and now autocomplete popup is displayed to see the a11y overlay
3. Open the Settings modal and disable the a11y overlay
4. Press ESC in the textarea again to see that no overlay is now displayed 
5. Check that the value is persisted in the local storage

Flaky test runner 
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/2346

### Checklist
